packages/core/src/libraries/user.ts

```diff
@@ -338,6 +241,66 @@ export const createUserLibrary = (queries: Queries) => {
   const findUserSsoIdentities = async (userId: string) =>
     userSsoIdentities.findUserSsoIdentitiesByUserId(userId);
 
+  type ProvisionOrganizationsParams =
+    | {
+        /** The user ID to provision organizations for. */
+        userId: string;
+        /** The user's email to determine JIT organizations. */
+        email: string;
+        /** The SSO connector ID to determine JIT organizations. */
+        ssoConnectorId?: undefined;
+      }
+    | {
+        /** The user ID to provision organizations for. */
+        userId: string;
+        /** The user's email to determine JIT organizations. */
+        email?: undefined;
+        /** The SSO connector ID to determine JIT organizations. */
+        ssoConnectorId: string;
+      };
+
+  // TODO: If the user's email is not verified, we should not provision the user into any organization.
+  /**
+   * Provision the user with JIT organizations and roles based on the user's email domain and the
+   * enterprise SSO connector.
+   */
+  const provisionOrganizations = async ({
+    userId,
+    email,
+    ssoConnectorId,
+  }: ProvisionOrganizationsParams): Promise<readonly JitOrganization[]> => {
+    const userEmailDomain = email?.split('@')[1];
```

Comment:
nit: use a util function for better debugging on error.

Reply:
can you elaborate this?

Reply:
discussed offline